common/proxyMiddleman_windows.go

```diff
@@ -0,0 +1,30 @@
+package common
+
+import (
+	"github.com/mattn/go-ieproxy"
+	"golang.org/x/net/http/httpproxy"
+	"net/http"
+	"net/url"
+)
+
+func proxyMiddleman() func(req *http.Request) (i *url.URL, e error) {
+	conf := ieproxy.GetConf()
```

Comment:
I've just realize something: GetConf only seems to cover two of the three cases that Windows supports. Windows supports naming a pac file (which is supported here by conf.Automatic in the return value), by static values (supported by conf.Static) AND by telling Windows "I'm not telling you anything. Just find the proxy for me all by yourself (using DNS and/or DHCP).".   You can see all three options in Window's Proxy Settings window.  But... the last one, of totally automatic config, is not covered by the existing code from the mattn library!  I previously assumed it was, but when I read the code in pac_windows.go I see that its only doing the option when it already has the URL of a pac script. It's not doing the DNS/DHCP option.  I think we need to cover that, otherwise we are not doing everythign that Windows does.

BUT, before going any further on this specific issue, we should figure out a way to TEST with that totally automatic config - since only with some tests setup will we be able to confirm whether we have a problem (I'm fairly sure we do) and whether we can fix it (which is what we most need testability for).